[4.5] AS support to the [cdr][field] default setting

daniel-lucio opened the pull request:
if you put a default setting like

_SESSION[cdr][field][array] = xml_cdr_uuid as CDR

it allows you to present a column with an alternative name. 

The best use case is letting users see their CALL UUID (when reporting a call) without letting them view the call detail. Very handy to do support.

daniel-lucio committed:
Update xml_cdr_inc.php

daniel-lucio committed:
Update xml_cdr.php

markjcrane commented:
If you want it committed provide an example use case that shows why it has value.

daniel-lucio commented:
> The best use case is letting users see their CALL UUID (when reporting a call) without letting them view the call detail. Very handy to do support.

![Screenshot_20210407_230049](https://user-images.githubusercontent.com/3492858/113962138-20eb3800-97f5-11eb-9b65-00e128064ad8.png)

When someone wants to report a call, it is more accurate to specify the UUID rather than date/caller/callee. Remember the date (in the CDR) is displayed in the local timezone which is very easy to mess up.

I also will ask you to start asking everybody for a use case.

daniel-lucio committed:
2 Bugfixes

1. My patch had a bug (fixed now). It was passing the INSERT (field AS alias, field1, field2, field3)  which is wrong. Fixed.

While I was fixing this, I discovered that the $this->field value is vulnerable to duplicate keys. If someone puts in the $_SESSION[cdr][field] default setting a field that is already added it will end with an SQL query like INSER (field2, field1, field2, field3). Postgresql and Mysql/mariadb won't work with duplicated fields in the insert query.
